modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/GcalKeywordReader.scala

```diff
@@ -32,16 +32,22 @@ object GcalKeywordsReaderImpl extends GcalKeywordReader {
   def getFilter: SeqAction[Option[String]] = GcalEpics.instance.filter
 
   def getLamp: SeqAction[Option[String]] = {
-    val ar   = GcalEpics.instance.lampAr().filter(_ == BinaryOnOff.On) *> "Ar".some
-    val cuAr = GcalEpics.instance.lampCuAr().filter(_ == BinaryOnOff.On) *> "CuAr".some
-    val ir   = GcalEpics.instance.lampIr().filter(_ == BinaryOnOff.On) *> "IR".some
-    val qh   = GcalEpics.instance.lampQH().filter(_ == BinaryOnOff.On) *> "QH".some
-    val thAr = GcalEpics.instance.lampThAr().filter(_ == BinaryOnOff.On) *> "ThAr".some
-    val xe   = GcalEpics.instance.lampXe().filter(_ == BinaryOnOff.On) *> "Xe".some
-
-    // Order is important to preserve alphabetical order
-    List(ar, cuAr, ir, qh, thAr, xe).flatten.mkString(" + ")
+    def ar   = GcalEpics.instance.lampAr().filter(_ == BinaryOnOff.ON) *> "Ar".some
+    def cuAr = GcalEpics.instance.lampCuAr().filter(_ == BinaryOnOff.ON) *> "CuAr".some
+    def ir   = GcalEpics.instance.lampIr().map{
+      case BinaryOnOff.ON => "IRhigh"
+      case _              => "IRlow"
+    }
+    def qh   = GcalEpics.instance.lampQH().filter(_ == BinaryOnOff.ON) *> "QH".some
+    def thAr = GcalEpics.instance.lampThAr().filter(_ == BinaryOnOff.ON) *> "ThAr".some
+    def xe   = GcalEpics.instance.lampXe().filter(_ == BinaryOnOff.ON) *> "Xe".some
+
+    Stream(ar, xe, cuAr, thAr, qh, ir).flatten.headOption
   }
```

Comment:
Ok, Shane's argument convinced me. But something worries me. Is that conversion the same that allows us to `flatMap` a `List` with a function that produce `Option`, and have as a result a `List` of only the inner values of the `Some`s, with the `None`s removed? Because if that is the case, we need to change the code in several parts.